pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -79,6 +80,20 @@ func (o Operation) String() string {
 	}
 }
 
+func (o *Operation) Unmarshal(v string) error {
```

Comment:
I was hoping this would be possible without the pointer, but I guess not with string values.

Considering this, maybe change this to a `NewOperation(v string) error`? It's awkward having to create a pointer first just to unmarshal.

Reply:
Notice that I didn't actually have to create a pointer:

```
		var op bpModel.Operation
		err := op.Unmarshal(req.Operation)
```

I ran it through a debugger and the underlying value changes as expected.

It's actually more awkward to create a pointer-to-int in a `NewOperation()` function than to work with an Operation instance in the example I gave.

Reply:
Huh ok, interesting. I was not aware this was possible. TIL!

The only problem my suggestion was meant to solve is having to create a pointer var, so if that's a non-issue we can disregard my suggestion.